Makefile

```diff
@@ -491,6 +493,15 @@ CXXFLAGS += $(WARNING_FLAGS) -I. -I./include $(PLATFORM_CXXFLAGS) $(OPT) -Woverl
 
 LDFLAGS += $(PLATFORM_LDFLAGS)
 
+ifeq ($(PLATFORM),OS_MACOSX)
+ifneq (,$(findstring -arch arm64,$(CXXFLAGS)))
+CFLAGS := $(filter-out -march=native -DHAVE_SSE42 -DROCKSDB_JEMALLOC -DJEMALLOC_NO_DEMANGLE $(JEMALLOC_INCLUDE) $(JEMALLOC_INCLUDE),$(CFLAGS))
+CXXFLAGS := $(filter-out -march=native -DHAVE_SSE42 -DROCKSDB_JEMALLOC -DJEMALLOC_NO_DEMANGLE $(JEMALLOC_INCLUDE) $(JEMALLOC_INCLUDE),$(CXXFLAGS))
+LDFLAGS := $(filter-out -ljemalloc,$(LDFLAGS))
+JAVA_LDFLAGS := $(filter-out -ljemalloc,$(JAVA_LDFLAGS))
+endif
+endif
+
```

Comment:
This seems like it makes more sense in build_detect_platform (with a comment).  Alternatively, why not move this earlier in the Makefile and just set DISABLE_JEMALLOC=1 (similar to ASAN and UBSAN).

Reply:
Good remark! The thing is that `build_detect_platform` assumes that the target OS/architecture will be the host OS/architecture. But when building for Mac, we might target a different architecture (arm64) than the one we're running on (x86_64). When that's the case, we need to remove flags such as `-march=native`. Since `build_detect_platform` doesn't know the actual target architecture, it seems that this operation can only be made in the Makefile (unless we pass it some option...)

So I updated the PR and went with your alternate solution: I moved the block earlier in the `Makefile` and used DISABLE_JEMALLOC. Thanks!